src/unitxt/metric_utils.py

```diff
@@ -345,6 +346,105 @@ def prepare(self):
 )
 
 
+class GlobalScores(dict):
+    @property
+    def main_score(self):
+        self["score"]
+
+    @property
+    def main_score_name(self):
+        self["score_name"]
+
+    def to_df(self):
+        """Transforms a dictionary of results into a pandas dataframe.
+
+        Transforms a dictionary of results into a dataframe with score_name as the index,
+        and columns for score, ci_low, and ci_high. Handles cases where confidence intervals are missing.
+
+        Returns:
+            pd.DataFrame: A dataframe with the extracted information, indexed by score_name.
+        """
+        import pandas as pd
+
+        rows = []
+
+        # Extract data based on score names
+        for key, value in self.items():
+            if (
+                key.endswith("_ci_low")
+                or key.endswith("_ci_high")
+                or key == "score_name"
+            ):
+                continue  # Skip confidence interval keys for now
+
+            if isinstance(value, (int, float)):  # Only consider numerical scores
+                score_name = key
+                ci_low = self.get(f"{key}_ci_low", None)
+                ci_high = self.get(f"{key}_ci_high", None)
+
+                rows.append(
+                    {
+                        "score_name": score_name,
+                        "score": value,
+                        "ci_low": ci_low,
+                        "ci_high": ci_high,
+                    }
+                )
+
+        df = pd.DataFrame(rows)
+        return df.set_index("score_name")
+
+
+class InstanceScores(list):
+    def to_df(self):
+        """Transforms a list of instance results into a pandas dataframe.
+
+        Returns:
+            pd.DataFrame: A dataframe with the extracted information, indexed by score_name.
+        """
+        from pandas import DataFrame
+
+        return DataFrame(self)
```

Comment:
We need to consider how to handle more complex fields like task data.

Reply:
I think task dataxshould be broken to its component.  We should change the example to use thus instead if print dice.